modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentResourceImpl.java

```diff
@@ -862,6 +867,51 @@ private StructuredContent _addStructuredContent(
 					structuredContent)));
 	}
 
+	private boolean _containsI18nMap(ContentField[] contentFields) {
+		if (ArrayUtil.isEmpty(contentFields)) {
+			return false;
+		}
+
+		for (ContentField contentField : contentFields) {
+			if (MapUtil.isNotEmpty(contentField.getContentFieldValue_i18n())) {
+				return true;
+			}
+
+			ContentField[] nestedFields = contentField.getNestedContentFields();
```

Comment:
@ak-ragnor, you can call here directly 

`_containsI18nMap(contentField.getNestedContentFields()); `

And avoid the next if